lib/state/services.dart

```diff
@@ -36,10 +36,10 @@ class ServiceRepository {
   /// Global service for performing HTTP calls.
   final HttpService http;
 
-  /// Global service for interacting with shared preferences.
-  final SharedPreferencesService sharedPreferences;
+  /// Global service for interacting with storage.
+  final StorageProvider storage;
```

Comment:
We don't need it yet (as the terms and conditions acceptance is global), but I think we should place the `StorageProvider` as a network service as most interactions will be parameterized by the currently active network. Just something to keep in mind, but not something that must be done in this PR.

Reply:
I would strongly prefer a single datastore with the network etc. being a field that we can filter on. We can add network-specific facade that does this filtering (unsure it that's what you meant @orhoj).

Reply:
As described below refactoring has been done into using a single storage provider and then possible network filterings done in repositories.